src/ConflictResolver/AutoIncrementConflictResolver.php

```diff
@@ -40,20 +42,28 @@ public function __construct(AdapterInterface $adapter)
      */
     public function resolveConflict(UriContext $uriContext)
     {
-        $this->inc = 0;
+        $this->index = 0;
+        $candidateUri = $uri = $uriContext->getUri();
 
-        $uri = $uriContext->getUri();
-        $candidateUri = $this->incrementUri($uri);
-
-        while ($route = $this->adapter->findRouteForUri($candidateUri, $uriContext)) {
+        while ($this->isUriConflicting($candidateUri, $uriContext)) {
             $candidateUri = $this->incrementUri($uri);
         }
 
         return $candidateUri;
     }
 
+    /**
+     * Tell if the given URI for the given context is conflicting with another
+     * route.
+     */
+    protected function isUriConflicting($uri, UriContext $uriContext)
+    {
+        return null !== $uriContext->getCollection()->getAutoRouteByUri($uri)
+            || null !== $this->adapter->findRouteForUri($uri, $uriContext);
```

Comment:
In the long run it might be better to delegate this responsiblity to a `ConflictChecker` or similar, f.e. `$conflictChecker->isConflicting($uriContext)` and pass it as a method argument (most if not all conflict resolvers would need to do the above check).